### ISSUE TITLE
Add option to ProdExcludesCheckMojo to ignore specific items in diff checks

### ISSUE THREAD

**Issue by jamesnetherton** [l2x6/cq-maven-plugin#318]
For example, building on Mac OS often results in the dependency tree being subtly different due to OS specific dependencies or profiles getting activated.

E.g with brotli4j:

```
Execution check-excludes of goal org.l2x6.cq:cq-prod-maven-plugin:4.4.5:prod-excludes-check failed: File [product/src/main/generated/transitive-dependencies-all.txt] is not in sync with product/src/main/resources/camel-quarkus-product-source.json:

Changed content at line 35:
expecting:
   ["com.aayushatharva.brotli4j:native-osx-aarch64"]
but was:
   ["com.aayushatharva.brotli4j:native-linux-x86_64"]

Missing content at line 836:
   ["net.openhft:affinity"]
```

**Comment by ppalaga** [comment]
I wonder whether ignoring is what we really want here. Yes, it would allow to pass `prod-excludes-check` on Mac, but the result of `prod-excludes` would still be incorrect. If folks on Mac merge it, we may have a problem. If possible we should try to force the resolver into thinking that it runs on Linux. It will be hard if the platform specific artifacts are added through a platform specific maven profile. Those ones are enabled based on `os.*` system properties, which might be hard/dangerous to fake. I wonder whether we shouldn't better introduce some sort of classifier replacement for well known suspects?

**Comment by jamesnetherton** [comment]
> I wonder whether we shouldn't better introduce some sort of classifier replacement for well known suspects?

Yes, agreed. I think that would work.

**Comment by jamesnetherton** [comment]
Fixed in 4.4.11 / 4.5.0.
